Add more comprehensive and crisp favicons

grossbart opened the pull request:
The previous favicons were not optimized for small sizes and were quite blurry. We now also provide a larger palette of icons for use-cases like adding a link to a home screen or tile.

I created these for Pursuit and thought that they are probably best kept here. I tried to come up with an approach that would work better for an open source project and that doesn't use Adobe file formats, but I did not get anything to work that I can run on my machine easily. But this remains an issue and we should use source files that can be edited on all platforms …

grossbart committed:
Add more comprehensive and crisp favicons

The previous favicons were not optimized for small sizes and were quite blurry. We now also provide a larger palette of icons for use-cases like adding a link to a home screen or tile.

hdgarrood commented on favicon/README.md:
```diff
@@ -0,0 +1,54 @@
+
+## Files used
+
+* `favicon.ico` – Used by IE, and also by some other browsers if we are not careful.
```
Can you clarify what "if we are not careful" means?

hdgarrood commented on favicon/README.md:
```diff
@@ -0,0 +1,54 @@
+* `mstile-150x150.png` – For Windows 8 / IE11.
+* `mstile-310x310.png` – For Windows 8 / IE11.
+* `mstile-310x150.png` – For Windows 8 / IE11.
+* `apple-touch-icon.png` – For iOS home screens.
```
Could you please add `manifest.json`, `safari-pinned-tab.svg`, and `browserconfig.xml` to this list?

hdgarrood commented on favicon/README.md:
```diff
@@ -0,0 +1,54 @@
+* `mstile-310x150.png` – For Windows 8 / IE11.
+* `apple-touch-icon.png` – For iOS home screens.
+
+These images were generated by [Real Favicon Generator](https://realfavicongenerator.net) on 2017-01-07. The following source files taken from the [Purescript Logo](https://github.com/purescript/logo) repository were used:
```
Probably don't need to specify where the source files came from since they are in the same directory.

hdgarrood commented on favicon/README.md:
```diff
@@ -0,0 +1,54 @@
+# Purescript Favicon
```
Oh also, capital S in PureScript please :)

grossbart committed:
Additional descriptions and typo fixes

grossbart commented:
Better? The worst typo has certainly been fixed: it's now PureScript everywhere ;)

grossbart committed:
Fix manifest files

Some information was specific to Pursuit, which doesn't make sense. Also, the generated manifest.json had a problem with escaped slash characters. And browserconfig.xml now references all the images, which was also only fixed in the Pursuit repo.

grossbart committed:
Fix icon path

Oh dear, what a lame typo …

hdgarrood reviewed (approved):
Looks good to me! Does anyone else want to review? If not I'll go ahead and merge.

paf31 commented:
👍 LGTM! Thanks @naehrstoff.

hdgarrood commented:
Thanks!
